log-collector-script/eks-log-collector.sh

```diff
@@ -0,0 +1,566 @@
+  echo ""
+  echo "   --ignore_metrics Variable To ignore prometheus metrics collection; Pass this flag if DISABLE_METRICS enabled on CNI"
+  echo ""
+  echo "   --help  Show this help message."
+  echo ""
+}
+
+parse_options() {
+  local count="$#"
+
+  for i in $(seq "${count}"); do
+    eval arg="\$$i"
+    param="$(echo "${arg}" | awk -F '=' '{print $1}' | sed -e 's|--||')"
+    val="$(echo "${arg}" | awk -F '=' '{print $2}')"
+
+    case "${param}" in
+      mode)
+        eval "${param}"="${val}"
+        ;;
+      ignore_introspection)
+        eval "${param}"="${val}"
+        ;;
+      ignore_metrics)
+        eval "${param}"="${val}"
+        ;;
+      help)
+        help && exit 0
+        ;;
+      *)
+        echo "Parameter not found: '$param'"
+        help && exit 1
+        ;;
+    esac
+  done
+}
+
+ok() {
+  echo
+}
+
+try() {
+  local action=$*
+  echo -n "Trying to $action... "
+}
+
+warning() {
+  local reason=$*
+  echo -e "\n\n\tWarning: $reason "
+}
+
+die() {
+  echo -e "\n\tFatal Error! $* Exiting!\n"
+  exit 1
+}
+
+is_root() {
+  if [[ "$(id -u)" -ne 0 ]]; then
+    die "This script must be run as root!"
+  fi
+}
+
+check_required_utils() {
+  for utils in ${REQUIRED_UTILS[*]}; do
+    # if exit code of "command -v" not equal to 0, fail
+    if ! command -v "${utils}" >/dev/null 2>&1; then
+      die "Application \"${utils}\" is missing, please install \"${utils}\" as this script requires it, and will not function without it."
+    fi
+  done
+}
+
+version_output() {
+  echo -e "\n\tThis is version ${PROGRAM_VERSION}. New versions can be found at ${PROGRAM_SOURCE}\n"
+}
+
+systemd_check() {
+  if  command -v systemctl >/dev/null 2>&1; then
+      INIT_TYPE="systemd"
+    else
+      INIT_TYPE="other"
+  fi
+}
+
+create_directories() {
+  # Make sure the directory the script lives in is there. Not an issue if
+  # the EKS AMI is used, as it will have it.
+  mkdir --parents "${PROGRAM_DIR}"
+  
+  # Common directors creation 
+  for directory in ${COMMON_DIRECTORIES[*]}; do
+    mkdir --parents "${COLLECT_DIR}"/"${directory}"
+  done
+}
+
+get_instance_metadata() {
+  readonly INSTANCE_ID=$(curl --max-time 3 --silent http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
+  echo "${INSTANCE_ID}" > "${COLLECT_DIR}"/system/instance-id.txt
+}
+
+is_diskfull() {
+  local threshold
+  local result
+
+  # 1.5GB in KB
+  threshold=1500000
+  result=$(df / | grep --invert-match "Filesystem" | awk '{ print $4 }')
+
+  # If "result" is less than or equal to "threshold", fail.
+  if [[ "${result}" -le "${threshold}" ]]; then
+    die "Free space on root volume is less than or equal to $((threshold>>10))MB, please ensure adequate disk space to collect and store the log files."
+  fi
+}
+
+cleanup() {
+  rm --recursive --force "${COLLECT_DIR}" >/dev/null 2>&1
+}
+
+init() {
+  check_required_utils
+  version_output
+  is_root
+  systemd_check
+  get_pkgtype
```

Comment:
Can we log the arguments as well?

Reply:
[Added](https://github.com/awslabs/amazon-eks-ami/blob/f6155ab958b630d7f8a45a13787d0dd2073481a5/log-collector-script/eks-log-collector.sh#L223)